### PR TITLE
Shortcode reversal for iframe shortcode

### PR DESCRIPTION
**Pull request by danielbachhuber**
See #99

### CODE REVIEW

File: inc/shortcodes/class-iframe.php [hunk 1]
```diff
@@ -49,6 +49,29 @@ public static function get_whitelisted_iframe_domains() {
...
+	public static function reversal( $content ) {
+
+		if ( $iframes = self::parse_iframes( $content ) ) {
+			$whitelisted_iframe_domains = static::get_whitelisted_iframe_domains();
+			$replacements = array();
+			foreach ( $iframes as $iframe ) {
+				if ( ! in_array( parse_url( $iframe->src_force_protocol, PHP_URL_HOST ), $whitelisted_iframe_domains ) ) {
+					continue;
+				}
+				$replacements[ $iframe->original ] = '[' . self::get_shortcode_tag() . ' src="' . esc_url_raw( $iframe->attrs['src'] ) . '"]';
```
Comment (danielbachhuber) [inline inc/shortcodes/class-iframe.php]: @goldenapples Do you think there's rationale for persisting iframe attributes too? If so, which, and how should we go about it?

**Comment by danielbachhuber** [comment]
@goldenapples #reviewmerge

File: inc/shortcodes/class-iframe.php (hunk 1 above)
Comment (goldenapples) [inline inc/shortcodes/class-iframe.php]: Hmm. Most of the attributes that we'll see will likely be fairly common styling options (`frameborder`, `border`, `scrolling`, `marginwidth`, `marginheight`) that we could store as a single attribute like 'iframe-style'.

The other attributes could probably be dealt with by a whitelist for now: just expose `id`, `class`, and maybe any others that are obvious... 

Persisting additional attributes might have to wait until https://github.com/fusioneng/Shortcake/issues/425 is fixed. Once that issue can be addressed, there wouldn't be any problem with taking any attributes that pass a `wp_kses_attr_check` and putting them in the shortcode, knowing that they won't be editable through the UI.

Comment (danielbachhuber) [inline inc/shortcodes/class-iframe.php]: Actually, I want to go with what we have for now — KISS. We can solve the persist attributes problem when we have it.

Comment (goldenapples) [inline inc/shortcodes/class-iframe.php]: Sounds good. :ship:

**Pull request merged**